mmengine/runner/runner.py

```diff
@@ -349,19 +350,38 @@ def __init__(
         self._randomness_cfg = randomness
         self.set_randomness(**randomness)
 
-        if experiment_name is not None:
-            self._experiment_name = f'{experiment_name}_{self._timestamp}'
-        elif self.cfg.filename is not None:
+        if experiment_name is None and self.cfg.filename is not None:
             filename_no_ext = osp.splitext(osp.basename(self.cfg.filename))[0]
-            self._experiment_name = f'{filename_no_ext}_{self._timestamp}'
+            experiment_name = f'{filename_no_ext}'
+
+        self._experiment_name = experiment_name
+        self._experiment_id = self._timestamp
+
+        if work_dir is None:
+            if 'work_dir' in self.cfg and self.cfg.work_dir is not None:
+                work_dir = self.cfg.work_dir
+            elif self._experiment_name is not None:
+                if auto_distinguish_experiment:
+                    work_dir = osp.join('./work_dirs', self._experiment_name,
+                                        self._experiment_id)
+                else:
+                    work_dir = osp.join('./work_dirs', self._experiment_name)
```

Comment:
Should `auto_distinguish_experiment` be the outer `if else`, and  `work_dir`, `log_dir` is set by different branch?

Reply:
emmmm……I think the current way is better, if we move auto_distinguish_experiment to outer if else, we will get so many duplicated lines.

Reply:
Or we can accept a little bit of logic modification as this. How about this?